compiler/prelude/types.js

```diff
@@ -150,7 +150,16 @@ var $newType = (size, kind, string, named, pkg, exported, constructor) => {
                     }), "$");
                 };
                 typ.copy = (dst, src) => {
-                    $copyArray(dst, src, 0, 0, src.length, elem);
+                    if (src.length === undefined) {
+                        // copy from a slice, the slice may be bigger but not smaller than the array
+                        if (src.$length < dst.length) {
+                            $throwRuntimeError("cannot convert slice with length "+src.$length+" to array or pointer to array with length "+dst.length);
+                        }
+                        $copyArray(dst, src.$array, 0, 0, Math.min(src.$length, dst.length), elem);
```

Comment:
It just occurred to me that `Math.min(src.$length, dst.length)` could simplified to just `dst.length`, since we know `dst.lenght` is less than or equal to the `src.$length`. I'll push up that fix in a second.

Reply:
Fixed